Marlin/src/module/stepper.cpp

```diff
@@ -2420,6 +2427,7 @@ hal_timer_t Stepper::block_phase_isr() {
         // step_rate to timer interval and steps per stepper isr
         interval = calc_multistep_timer_interval(acc_step_rate << oversampling_factor);
         acceleration_time += interval;
+        deceleration_time = 0; // Reset since we're doing acceleration first.
```

Comment:
I think this should be `deceleration_time = interval / 2;` for the case where there are no cruise steps.

Reply:
0 (which means 0 speed change) is the value that results in the smoothest transition between full acceleration and full deceleration. We get full accel -> 0 -> full decel (also visible on the logic analyzer capture).